### PR TITLE
fix #297719 : [Linux AppImage] System printers not available in print…

### PR DESCRIPTION
**Pull request by AntonioBL**
… dialogue

Resolves: https://musescore.org/en/node/297719

Linuxdeploy, with qt-plugin, used to deploy the AppImage, does not include at the moment the library libcupsprintersupport.so which is needed for printing support, so we should manually include it.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [n/a] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by AntonioBL** [comment]
This should solve the problem for the 64bit AppImage. I don't know if the problem appears also for the 32bit one; I don't think linuxdeploy is used for the 32bit AppImage, so libcupsprintersupport.so should be added by the old (manually made) script.

**Comment by AntonioBL** [comment]
There are probably other ways to solve this problem.
Almost one year ago I tried with EXTRA_QT_PLUGINS variable, but it didn't seem to work in this case.

**Comment by MarcSabatella** [comment]
Thanks for doing this!  I wonder, any chance that it helps with getting the default page size, which also doesn’t work on Linux?

**Comment by AntonioBL** [comment]
I had forgotten to make the new directory, now I fixed it.
I also once again verified locally that EXTRA_QT_PLUGINS is not working in this case; I don't know why.

**Comment by AntonioBL** [comment]
From what I can see, @shoogle is almost ready with a PR covering also this case:
https://github.com/shoogle/MuseScore/tree/appimage-libraries

**Comment by shoogle** [comment]
@AntonioBL, that's correct. Nothing wrong with what you have done here, but I prefer to solve things in a more general way if possible.

My PR also does other things so feel free to borrow the relevant lines from it and add them to this PR. The changes might get merged sooner that way.

BTW, if adding the library solves the issue (which I cannot test as I don't have a printer) then it should be reported to https://github.com/linuxdeploy/linuxdeploy-plugin-qt/issues

**Comment by AntonioBL** [comment]
I think it would be better to merge your (future) PR, which contains also the solution for other missing libraries (and the solution for the case in which /sbin is not in the PATH).

For what concerns linuxdeploy-plugin-qt, it actually tries to include the printsupport, see
https://travis-ci.org/musescore/MuseScore/jobs/634744261#L3971
but there is no output, and no Qt plugin libraries deployed.

**Comment by dmitrio95** [comment]
So should this PR get merged or should we better wait for another solution?

**Comment by AntonioBL** [comment]
Not an easy answer.
This PR works at the moment. However, @shoogle 's work is more general and solves this bug (in basically a similar way) and another couple of bugs (missing libraries and missing folders in PATH).
See https://github.com/shoogle/MuseScore/commits/appimage-libraries
In my opinion, in view of merging, his work on the branch is a better candidate than this. I think he is only waiting for feedback if it solves all the problem and then should be sending a PR.
[ Only if 3.4 release is in so short time that a PR from that branch is not ready then this PR can be considered as a temporary "patch". However, if this is merged, then that branch has to be rebased.
Long story short: if possible, I would wait for shoogle's PR. ]

**Comment by Jojo-Schmitz** [comment]
Which is in #5607 now ;-)

**Comment by dmitrio95** [comment]
OK, let's use that one instead then.
